src/cmd/src/cli/export.rs

```diff
@@ -232,19 +253,57 @@ impl Export {
         show_type: &str,
         catalog: &str,
         schema: &str,
-        table: &str,
+        table: Option<&str>,
     ) -> Result<String> {
-        let sql = format!(
-            r#"SHOW CREATE {} "{}"."{}"."{}""#,
-            show_type, catalog, schema, table
-        );
-        let result = self.database_client.sql_in_public(&sql).await?;
-        let records = result.context(EmptyResultSnafu)?;
-        let Value::String(create_table) = &records[0][1] else {
+        let sql = match table {
+            Some(table) => format!(
+                r#"SHOW CREATE {} "{}"."{}"."{}""#,
+                show_type, catalog, schema, table
+            ),
+            None => format!(r#"SHOW CREATE {} "{}"."{}""#, show_type, catalog, schema),
+        };
+        let records = self
+            .database_client
+            .sql_in_public(&sql)
+            .await?
+            .context(EmptyResultSnafu)?;
+        let Value::String(create) = &records[0][1] else {
             unreachable!()
         };
 
-        Ok(format!("{};\n", create_table))
+        Ok(format!("{};\n", create))
+    }
+
+    async fn export_create_database(&self) -> Result<()> {
+        let timer = Instant::now();
+        let db_names = self.get_db_names().await?;
+        let db_count = db_names.len();
+        for schema in db_names {
+            let db_dir = self.catalog_path().join(format!("{schema}/"));
+            tokio::fs::create_dir_all(&db_dir)
+                .await
+                .context(FileIoSnafu)?;
+            let file = db_dir.join("create_database.sql");
+            let mut file = File::create(file).await.context(FileIoSnafu)?;
+            match self
+                .show_create("DATABASE", &self.catalog, &schema, None)
+                .await
+            {
+                Err(e) => {
+                    error!(e; r#"Failed to export database schema "{}"."{}""#, self.catalog, schema)
```

Comment:
Is it better to change the behavior to "fail-fast" here? If something goes wrong for showing one database, the successive databases' "show" might be in vain: user has to execute the export again for the integrity of data.

Reply:
Good catch, I am going to make all `show create database` and `show create table` to be fail-fast,